ssl-config/build.gradle

```diff
@@ -21,3 +29,45 @@ task generateCerts(type:Exec) {
 
 tasks.check.dependsOn(javadoc)
 tasks.test.dependsOn(generateCerts)
+
+dockerTestRunner {
+    // missing BouncyCastle but contains Sun libraries
+    dockerRunner name: 'openjdk-7', dockerFile: file('src/test/resources/openjdk-7.dockerfile')
+
+    // missing BouncyCastle and Sun libraries
+    dockerRunner name: 'ibmjdk-8', dockerFile: file('src/test/resources/ibmjdk-8.dockerfile')
+
+    // run only the PKCS1 tests and exclude BouncyCastle runtime dependency
+    testConfig = {
+        include('**/pkcs1/*Tests*')
+        classpath = classpath - getFilesForDependency('testRuntime', bcpkix)
+    }
+
+    if (System.env.CIRCLE_TEST_REPORTS) {
+        runArgs.put('-v', "${System.env.CIRCLE_TEST_REPORTS}:/reports")
+        runArgs.put('-e', 'CIRCLE_TEST_REPORTS=/reports')
+    }
+}
+
+/**
+ * Returns a FileCollection that contains all of the files provided by the provided dependency in the given
+ * configuration.
+ */
+private FileCollection getFilesForDependency(String configName, String dependencyString) {
+    Set<ResolvedDependency> firstLevel = project
+            .configurations
+            .getByName(configName)
+            .resolvedConfiguration.getFirstLevelModuleDependencies()
+
+    Dependency dependency = project.dependencies.create(dependencyString)
```

Comment:
Because the `dependencyString` can be any String that specifies a dependency (for example, in this case it's `org.bouncycastle:bcpkix-jdk15on:1.54`, but can be any valid dependency identifier String that Gradle can resolve), and I don't want to implement the logic of trying to parse out the name in all the different cases.